sympy/polys/matrices/tests/test_domainmatrix.py

```diff
@@ -440,3 +440,19 @@ def test_DomainMatrix_zeros():
     assert A.rep == SDM.zeros((1, 2), QQ)
     assert A.shape == (1, 2)
     assert A.domain == QQ
+
+def test_DomainMatrix_getitem():
+    dm = DomainMatrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]], (3, 3), ZZ)
+    assert dm[::-1, :] == dm
+    assert dm[1:,:-3] == DomainMatrix([[4], [7]], (2, 1), ZZ)
```

Comment:
When constructing a DomainMatrix you need to pass in domain elements e.g. `ZZ(4)` rather than `4`.

Reply:
Sure.
Thanks.